From e605ad9679b583bf7e09afe00daf23e0dfa6c823 Mon Sep 17 00:00:00 2001
From: Akim Demaille <akim.demaille@gmail.com>
Date: Tue, 30 Oct 2018 06:55:47 +0100
Subject: build: fix use of gnulib Make variables

Reported by Kiyoshi Kanazawa.
http://lists.gnu.org/archive/html/bug-bison/2018-10/msg00048.html

* lib/local.mk (lib_libbison_a_LIBADD): Merge into...
* src/local.mk (src_bison_LDADD): here.
diff -wpruN '--exclude=*.orig' a~/THANKS a/THANKS
--- a~/THANKS	1970-01-01 00:00:00
+++ a/THANKS	1970-01-01 00:00:00
@@ -84,6 +84,7 @@ Juan Manuel Guerrero      juan.guerrero@
 Kees Zeelenberg           kzlg@users.sourceforge.net
 Keith Browne              kbrowne@legato.com
 Ken Moffat                zarniwhoop@ntlworld.com
+Kiyoshi Kanazawa          yoi_no_myoujou@yahoo.co.jp
 Laurent Mascherpa         laurent.mascherpa@epita.fr
 Lie Yan                   lie.yan@kaust.edu.sa
 Magnus Fromreide          magfr@lysator.liu.se
diff -wpruN '--exclude=*.orig' a~/lib/local.mk a/lib/local.mk
--- a~/lib/local.mk	1970-01-01 00:00:00
+++ a/lib/local.mk	1970-01-01 00:00:00
@@ -45,15 +45,6 @@ lib_libbison_a_SOURCES +=
   lib/path-join.h                               \
   lib/path-join.c
 
-lib_libbison_a_LIBADD +=                        \
-  $(ISNAND_LIBM)                                \
-  $(ISNANF_LIBM)                                \
-  $(ISNANL_LIBM)                                \
-  $(LDEXPL_LIBM)                                \
-  $(LDEXP_LIBM)                                 \
-  $(LIB_CLOCK_GETTIME)                          \
-  $(LIB_GETHRXTIME)
-
 # The Yacc compatibility library.
 if ENABLE_YACC
   lib_LIBRARIES = lib/liby.a
diff -wpruN '--exclude=*.orig' a~/src/local.mk a/src/local.mk
--- a~/src/local.mk	1970-01-01 00:00:00
+++ a/src/local.mk	1970-01-01 00:00:00
@@ -104,9 +104,19 @@ BUILT_SOURCES +=
   src/scan-gram.c                               \
   src/scan-skel.c
 
+# Although conceptually most of these guys would make more sense in the
+# definition of libbison, beware that they might expand as flags such as
+# `-lm`.  Keep them here.  Or use a Libtool convenience library.
 src_bison_LDADD =                               \
+  $(ISNAND_LIBM)                                \
+  $(ISNANF_LIBM)                                \
+  $(ISNANL_LIBM)                                \
+  $(LDEXPL_LIBM)                                \
+  $(LDEXP_LIBM)                                 \
   $(LIBINTL)                                    \
   $(LIBTHREAD)                                  \
+  $(LIB_CLOCK_GETTIME)                          \
+  $(LIB_GETHRXTIME)                             \
   lib/libbison.a
 
 
